### PR TITLE
Upgrade to dotnet core 3.1

### PR DESCRIPTION
**Pull request by live1206**
###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/docfx/pull/5394)

### CODE REVIEW

File: azure-pipelines-v3.yml [hunk 1]
```diff
@@ -17,10 +17,10 @@ jobs:
 
   # Install .NET Core sdk
   - task: UseDotNet@2
-    displayName: 'Install .NET Core sdk 2.2.300'
+    displayName: 'Install .NET Core sdk 3.1.100'
```
Comment (yufeih) [inline azure-pipelines-v3.yml]: consider extracting it to a variable: https://github.com/dotnet/docfx/pull/5397/files#diff-bee4d79cd349ddb0cb2b4b7d5a8bf3d3R10

File: src/Microsoft.Docs.Template/Microsoft.Docs.Template.csproj [hunk 2]
```diff
@@ -1,11 +1,17 @@
...
   <PropertyGroup>
     <TargetFramework>netstandard2.0</TargetFramework>
+    <AddRazorSupportForMvc>true</AddRazorSupportForMvc>
   </PropertyGroup>
 
   <ItemGroup>
     <PackageReference Include="Microsoft.AspNetCore.Mvc" Version="2.2.0" />
   </ItemGroup>
 
+  <ItemGroup>
```
Comment (yufeih) [inline src/Microsoft.Docs.Template/Microsoft.Docs.Template.csproj]: update `src/Directory.Build.props` , can then remove this section

File: src/docfx/docfx.csproj [hunk 3]
```diff
@@ -18,26 +18,31 @@
     <!-- To keep HTML intact during v3 migration, keep using HtmlAgilityPack 1.4.9 -->
     <PackageReference Include="HtmlAgilityPack" Version="1.4.9" />
     <PackageReference Include="LibGit2Sharp.NativeBinaries" Version="2.0.298" />
-    <PackageReference Include="Microsoft.ApplicationInsights" Version="2.11.0" />
+    <PackageReference Include="Microsoft.ApplicationInsights" Version="2.12.0" />
     <PackageReference Include="Microsoft.AspNetCore" Version="2.2.0" />
```
Comment (yufeih) [inline src/docfx/docfx.csproj]: This is still using ASPNET Core 2, interestingly APSNET Core 3 doesn't ship as NuGet package: https://docs.microsoft.com/en-us/aspnet/core/migration/22-to-30?view=aspnetcore-3.1&tabs=visual-studio#remove-obsolete-package-references

**Comment by live1206** [comment]
Recreated #5414 instead

**Pull request closed without merging**